### PR TITLE
v15 JSON  sparse input issue #87

### PR DESCRIPTION
**Pull request by josiahseaman**
This requires a new v15 format of JSON that is 32x smaller for B1phi1.  It happily simplifies some of our code.  Main testing needs to be the preservation of old features.  Test to make sure this is not noticably slower than master.

Can we profile javascript runtime?

image

### CODE REVIEW

File: src/ComponentRect.js [hunk 1]
```diff
@@ -1,45 +1,25 @@
...
-};
+import {sum} from "./utilities";
 
 export function compress_visible_rows(components) {
   /*Returns a Map with key of the original row number and value of the new, compressed row number.
    * Use this for y values of occupancy and LinkColumn cells.  */
-  let rows_present = find_rows_visible_in_viewport(components);
+  let all_visible = new Set();
+  for (let c of components) {
+    c.occupants.map((row) => all_visible.add(row));
```
Comment (AndreaGuarracino) [inline src/ComponentRect.js]: Is WebStorm not complaining that the vector returned by map() is not considered?

**Comment by lomereiter** [comment]
@josiahseaman modern browsers come with profiling tools built-in: https://developers.google.com/web/tools/chrome-devtools/evaluate-performance

File: src/ComponentRect.js (hunk 1 above)
Comment (lomereiter) [reply src/ComponentRect.js]: +1, looks like a `.forEach` candidate

Comment (josiahseaman) [reply src/ComponentRect.js]: I've just pushed a new commit with this fix.  Ready to merge?  @AndreaGuarracino  it's your call.

**Pull request merged**